### PR TITLE
Mac support fir for rabbitmq

### PR DESCRIPTION
**Pull request by alvagante**
This fixes the issues with rabbitmq, when used, as mentioned in https://github.com/sensu/sensu-puppet/pull/826/files#submit-review

Still I'm not sure on what should be the value ensure for MacOS and why it should have different values than the others.

### CODE REVIEW

**Comment by alvagante** [comment]
there are also some additions commits from a more recent master , but I suppose you'll have to squash everything in any case

**Comment by alvagante** [comment]
After a better look at what is done during provisioning on Vagrant, I think you just needed to have ensure = present when $::sensu::transport_type == 'rabbitmq'.
No need of special selector for MacOS

File: manifests/transport.pp [hunk 1]
```diff
@@ -5,13 +5,14 @@
 #
 class sensu::transport {
 
-  if $::sensu::transport_type != 'redis' {
+  if $::sensu::transport_type == 'redis'
+  or $::sensu::transport_type == 'rabbitmq' {
+    $ensure = 'present'
+  } else {
     $ensure = $::osfamily ? {
```
Comment (alvagante) [inline manifests/transport.pp]: This selector is probably not needed anymore if you add the or $::sensu::transport_type == 'rabbitmq' line.

**Comment by csoleimani** [comment]
Sorry I've been on vacation. Just merged your changes - looking at the Vagrantfile thing now. Did you get the USB error on trying vagrant up sensu-server?

**Pull request merged**